pymc/distributions/timeseries.py

```diff
@@ -292,37 +187,80 @@ def dist(cls, mu=0.0, sigma=1.0, *, init_dist=None, steps=None, **kwargs) -> at.
 
         return super().dist([mu, sigma, init_dist, steps], **kwargs)
 
-    def moment(rv, size, mu, sigma, init_dist, steps):
-        grw_moment = at.zeros_like(rv)
-        grw_moment = at.set_subtensor(grw_moment[..., 0], moment(init_dist))
-        # Add one dimension to the right, so that mu broadcasts safely along the steps
-        # dimension
-        grw_moment = at.set_subtensor(grw_moment[..., 1:], mu[..., None])
-        return at.cumsum(grw_moment, axis=-1)
-
-    def logp(
-        value: at.Variable,
-        mu: at.Variable,
-        sigma: at.Variable,
-        init_dist: at.Variable,
-        steps: at.Variable,
-    ) -> at.TensorVariable:
-        """Calculate log-probability of Gaussian Random Walk distribution at specified value."""
-
-        # Calculate initialization logp
-        init_logp = logp(init_dist, value[..., 0])
-
-        # Make time series stationary around the mean value
-        stationary_series = value[..., 1:] - value[..., :-1]
-        # Add one dimension to the right, so that mu and sigma broadcast safely along
-        # the steps dimension
-        series_logp = logp(Normal.dist(mu[..., None], sigma[..., None]), stationary_series)
-
-        return check_parameters(
-            init_logp + series_logp.sum(axis=-1),
-            steps > 0,
-            msg="steps > 0",
+    @classmethod
+    def ndim_supp(cls, *args):
+        return 1
+
+    @classmethod
+    def rv_op(cls, mu, sigma, init_dist, steps, size=None):
+        # If not explicit, size is determined by the shapes of mu, sigma, and init
+        if size is not None:
+            # we have all the information regarding size from users or .dist()
+            init_size = size
+        else:
+            # we infer size from parameters
+            init_size = at.broadcast_shape(mu, sigma, init_dist)
+
+        init_dist = change_rv_size(init_dist, init_size)[..., None]
+
+        innovation_dist = Normal.dist(mu[..., None], sigma[..., None], size=(*init_size, steps))
+
+        # TODO: extend for multivariate init
+        rv_out = at.cumsum(at.concatenate([init_dist, innovation_dist], axis=-1), axis=-1)
+        rv_out.tag.mu = mu
+        rv_out.tag.sigma = sigma
+        rv_out.tag.init_dist = init_dist.owner.inputs[0]
+        rv_out.tag.steps = steps
+
+        return rv_out
+
+    @classmethod
+    def change_size(cls, rv, new_size, expand=False):
+        if expand:
+            new_size = at.concatenate([new_size, rv.shape])
```

Comment:
I think the problem is here, size should not include the last dimension.
```suggestion
            new_size = at.concatenate([new_size, rv.shape[:-1])
```

Reply:
It works :) The `else` statement had to be removed altogether too

Reply:
Makes sense!